src/GafferModule/NameValuePlugBinding.cpp

```diff
@@ -0,0 +1,182 @@
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "boost/python.hpp"
+
+#include "NameValuePlugBinding.h"
+
+#include "GafferBindings/ValuePlugBinding.h"
+#include "Gaffer/NameValuePlug.h"
+
+#include "IECorePython/RunTimeTypedBinding.h"
+#include "IECorePython/ScopedGILRelease.h"
+
+using namespace boost::python;
+using namespace GafferBindings;
+using namespace Gaffer;
+
+namespace
+{
+
+class NameValuePlugSerialiser : public ValuePlugSerialiser
+{
+
+	public :
+
+		bool childNeedsConstruction( const Gaffer::GraphComponent *child, const Serialisation &serialisation ) const override
+		{
+			// The children will be created by the constructor output by repr
+			return false;
+		}
+
+		std::string constructor( const Gaffer::GraphComponent *graphComponent, const Serialisation &serialisation ) const override
+		{
+			return repr( static_cast<const NameValuePlug *>( graphComponent ), &serialisation );
+		}
+	
+
+		static std::string repr( const Gaffer::NameValuePlug *plug, const Serialisation *serialisation )
+		{
+			if( !plug->namePlug() || !plug->valuePlug<Plug>() )
+			{
+				throw IECore::Exception( "Cannot serialize: " + plug->fullName() + " - NameValuePlug must have name and value." );
+			}
+
+			std::string result = "Gaffer.NameValuePlug( ";
+			result += "\"" + plug->namePlug()->getValue() + "\", ";
```

Comment:
Shouldn't this be `namePlug()->defaultValue()` rather than `namePlug()->getValue()`?